script.js

```diff
@@ -0,0 +1,236 @@
+        const editTodoInputField = item.childNodes[1].childNodes[1].childNodes[3];
+        const description = item.childNodes[1].childNodes[1].childNodes[5];
+        const dragIcon = item.childNodes[1].childNodes[3];
+        const trashIcon = item.childNodes[1].childNodes[5];
+
+        this.showElements([editTodoInputField, trashIcon]);
+        this.hideElements([description, dragIcon]);
+
+        // listen for enter key press to edit item
+        editTodoInputField.addEventListener('keydown', (event) => {
+          if (event.key === 'Enter') {
+            updateText = editTodoInputField.value;
+            this.tasksArray.filter((task, taskIndex) => {
+              if (index === taskIndex) {
+                task.description = updateText;
+                this.updateTask(taskIndex, task);
+              }
+              return false;
+            });
+          }
+        });
+
+        // listen for trash icon clicked
+        trashIcon.addEventListener('click', () => {
+          const updatedArray = this.tasksArray.filter((task, taskIndex) => taskIndex !== index);
+          this.localCache(updatedArray);
+        });
+
+        // click event to hide the edit input field and trash icon
+        item.addEventListener('mouseleave', () => {
+          item.classList.remove('bg-yellow');
+          this.showElements([description, dragIcon]);
+          this.hideElements([editTodoInputField, trashIcon]);
+        });
+      });
+    });
+  }
+
+  clearAllCompleted = () => {
+    const uncompletedTasks = this.tasksArray.filter((task) => task.completed !== true);
+    this.btnClearAllCompleted.addEventListener('click', () => {
+      this.localCache(uncompletedTasks);
+    });
+  }
+
+  clearAllTasks = () => {
+    this.refreshIcon.addEventListener('click', () => {
+      this.refreshIcon.style.transform = 'rotate(360deg)';
+      this.localCache([]);
+    });
+  }
+
+  getTasksArray = () => {
+    const data = this.tasksArray;
+    return data;
+  }
+
+  hideElements = (args) => {
+    args.forEach((element) => {
+      element.style.display = 'none';
+    });
+  }
+
+  showElements = (args) => {
+    args.forEach((element) => {
+      element.style.display = 'flex';
+    });
+  }
+}
+
+const instance = new DataStore();
+export const showTasks = instance.readTasks();
+export const tasksArray = instance.getTasksArray();
+instance.addTask();
+instance.handleStatusChange();
+instance.clearAllCompleted();
+instance.editTask();
+instance.clearAllTasks();
+
+let dragItem = null;
+let dragItemIndex = null;
+let nextItemIndex = null;
+let nextElement = null;
+class Dragging {
+  constructor() {
+    this.todoTasks = tasksArray;
+  }
+
+  dragStart = (event) => {
+    dragItem = event.target;
+    dragItemIndex = +dragItem.id;
+    dragItem.classList.add('bg-white');
+  };
+
+  dragOver = (event) => {
+    event.preventDefault();
+    event.target.classList.add('drag-over');
+    const hoveredElement = event.target.parentNode;
```

Comment:
- [ ] The dragging class could be in another module for reusability and modularity

Reply:
Okay, thanks for the review. I will make the required changes.